### PR TITLE
chore: rename Go module as `github.com/ava-labs/libevm`

### PR DESCRIPTION
**Pull request by ARR4N**
## Why this should be merged

Completes renaming of the Go module to `github.com/ava-labs/libevm` to no longer clash with the true `github.com/ethereum/go-ethereum` and require a `replace` directive.

## How this works

The merge commit `d6d53f9` merges the `renamed-go-module` branch, resolves conflicts (imports only) and _also_ renames all other imports using the same `sed` regex/replace as the auto-renaming workflow. Completing the import renaming in a single commit made it self-contained and able to build and pass all tests.

The other commit makes all necessary CI changes to accomodate for the renaming. Of note, it removes use of the `libevm-base` tag and instead includes the commit hash in the relevant workflow; this allows for an atomic change of base commit at time of merging.

## How this was tested

Renaming of the module is confirmed by inspection of `go.mod`. Updating all internal import paths is confirmed by an updated linter that blocks modules, guaranteeing that the `ethereum/go-ethereum` module isn't imported.

The `libevm delta` workflow can also be inspected to see that the current diff doesn't show any module-name differences, while the diff against `main` does.

All regular CI (Go build + test) passes as expected.

### Merge checklist

- [x] Replace `a7f08d0` with a signed equivalent from #61
- [x] Clean commit history (even if force-push required) as this will be fast-forward merged
- [ ] Fast-forward merge on command line: DO NOT MERGE via GitHub UI

### CODE REVIEW

**Review by darioush — APPROVED** [review]

**Comment by ARR4N** [comment]
Blocked by #61 because [a7f08d0](https://github.com/ava-labs/libevm/pull/60/commits/a7f08d0e757d5a69d3e269c69dcea7e45bab97e3) wasn't signed 🤦 

Once that's merged, I'll substitute the above commit with a signed one from the updated workflow (e.g. [572b8ab](https://github.com/ava-labs/libevm/commit/572b8ab74eeb2afab3dbf353207d80145c2a1742), which is otherwise identical).

**Comment by ARR4N** [comment]
Note that the force push after review was to use a signed commit for the automated renaming. The [comparison](https://github.com/ava-labs/libevm/compare/4c8d4bde2c014c6a606abb72190e88f5099b2b8e..5c6635282d8dda1afe8117edbc0e1c3215cb6e94) shows that the push was a no-op with respect to the outcome—the only difference is the new renaming workflow to sign commits (i.e. the reason the push was needed) as reviewed and merged in #61.

**Pull request merged**
